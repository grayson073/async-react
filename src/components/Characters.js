import React, { Component } from 'react';
import Character from './Character';

const results = [
  {
    name: 'Frank',
    status: 'alive',
    species: 'dog',
    gender: 'male'
  },
  {
    name: 'Papi',
    status: 'alive',
    species: 'dog',
    gender: 'male'
  }
];

export default class Characters extends Component {

  state = {
    characters: []
  };
  
  componentDidMount() {
    this.setState({ characters: results });
  }
  
  render() {
    const listOfChars = this.state.characters.map(char => {
      return <Character character={char} key={char.name}/>;
    });

    return (
      <table>
        <tbody>
          {listOfChars}
        </tbody>
      </table>
    );
  }
}
