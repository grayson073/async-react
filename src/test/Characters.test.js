import React from 'react';
import renderer from 'react-test-renderer';
import Characters from '../components/Characters';

describe('Characters', () => {
  
  it('Renders correctly', () => {
    const characters = [
      {
        name: 'Frank',
        status: 'alive',
        species: 'dog',
        gender: 'male'
      },
      {
        name: 'Papi',
        status: 'alive',
        species: 'dog',
        gender: 'male'
      }
    ];

    const tree = renderer.create(
      <Characters characters={characters}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
