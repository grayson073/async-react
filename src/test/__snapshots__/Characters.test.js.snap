// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Renders correctly 1`] = `
<table>
  <tr>
    <td>
      Frank
    </td>
    <td>
      alive
    </td>
    <td>
      dog
    </td>
    <td>
      male
    </td>
  </tr>
  <tr>
    <td>
      Papi
    </td>
    <td>
      alive
    </td>
    <td>
      dog
    </td>
    <td>
      male
    </td>
  </tr>
</table>
`;
